app/HMS/Entities/Invite.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace HMS\Entities;
+
+use Carbon\Carbon;
+use HMS\Traits\Entities\SoftDeletable;
+use HMS\Traits\Entities\Timestampable;
+use LaravelDoctrine\ORM\Notifications\Notifiable;
+
+class Invite
+{
+    use Notifiable, SoftDeletable, Timestampable;
+
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var string email
+     */
+    protected $email;
+
+    /**
+     * @var string invite token
+     */
+    protected $inviteToken;
+
+    /**
+     * Invite constructor.
+     * Invite Token is auto generated.
+     *
+     * @param $email
+     * @return Invite
+     */
+    public function create($email)
+    {
+        $this->email = $email;
+        $this->inviteToken = hash_hmac('sha256', str_random(40), config('app.key'));
```

Comment:
Probs should reduce the collision space on this. Perhaps add the current timestamp into the data? i.e. str_random(40) + "" + $now